reana_job_controller/kubernetes_job_manager.py

```diff
@@ -157,10 +178,10 @@ def execute(self):
             for var, value in self.env_vars.items():
                 job_spec["containers"][0]["env"].append({"name": var, "value": value})
 
-        if REANA_KUBERNETES_JOBS_MEMORY_LIMIT:
-            job_spec["containers"][0]["resources"] = {
-                "limits": {"memory": REANA_KUBERNETES_JOBS_MEMORY_LIMIT,}
-            }
+        if self.kubernetes_memory_limit:
+            _set_job_memory_limit(job_spec, self.kubernetes_memory_limit)
+        elif REANA_KUBERNETES_JOBS_MEMORY_LIMIT:
+            _set_job_memory_limit(job_spec, REANA_KUBERNETES_JOBS_MEMORY_LIMIT)
```

Comment:
Minor: Perhaps we can follow the same pattern as `self.add_hostpath_volumes()`, e.g. `self.add_memory_limit()` which would handle the logic internally?